infoschema/tables.go

```diff
@@ -1828,6 +1843,112 @@ func dataForServersInfo() ([][]types.Datum, error) {
 	return rows, nil
 }
 
+func dataForTiDBClusterInfo(ctx sessionctx.Context) ([][]types.Datum, error) {
+	// Get TiDB servers info.
+	tidbNodes, err := infosync.GetAllServerInfo(context.Background())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	rows := make([][]types.Datum, 0, len(tidbNodes))
+	for _, node := range tidbNodes {
+		tp := "tidb"
+		name := fmt.Sprintf("%s-%d", tp, len(rows))
+		addr := fmt.Sprintf("%s:%d", node.IP, node.Port)
+		statusAddr := fmt.Sprintf("%s:%d", node.IP, node.StatusPort)
+		row := types.MakeDatums(
+			len(rows)+1,
+			tp,
+			name,
+			addr,
+			statusAddr,
+			node.Version,
+			node.GitHash,
+		)
+		rows = append(rows, row)
+	}
+
+	// Get PD servers info.
+	store := ctx.GetStore()
+	etcd, ok := store.(tikv.EtcdBackend)
+	if !ok {
+		return nil, errors.Errorf("%T not an etcd backend", store)
+	}
+	for i, addr := range etcd.EtcdAddrs() {
+		addr = strings.TrimSpace(addr)
+
+		// get pd version
+		url := fmt.Sprintf("http://%s%s", addr, pdapi.ClusterVersion)
+		resp, err := http.Get(url)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		pdVersion, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		terror.Log(resp.Body.Close())
+		version := strings.Trim(strings.Trim(string(pdVersion), "\n"), "\"")
+
+		// Get PD git_hash
+		url = fmt.Sprintf("http://%s%s", addr, pdapi.Status)
+		resp, err = http.Get(url)
```

Comment:
The request will always handle by PD leader, should also change PD?
![image](https://user-images.githubusercontent.com/26020263/68099304-b695e000-fefc-11e9-822a-18e71710f33b.png)

Reply:
Nice catch, seems PD should not redirect all request to leader.

Reply:
https://github.com/pingcap/pd/pull/1883